### PR TITLE
[CDC|Waiver] Adding more waivers

### PR DESCRIPTION
**Pull request by jeoongp**
- Misc paths are added
- W_FANOUT and W_INTERFACE items are added

Signed-off-by: Joshua Park <jeoong@google.com>

### CODE REVIEW

**Comment by msfschaffner** [comment]
I will take a look in a bit.

File: hw/top_earlgrey/cdc/cdc_waivers.w_fanout.tcl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Verix CDC waiver file
+
+set_rule_status -rule {W_FANOUT} -expression {Driver =~ "*u_lc_ctrl*.u_prim_lc_sender_escalate_en*"} -status {Waived} -comment {No Reconvergence issue. Each IP handles Escalate En individually; set_rule_status run by  user root on host smart-dis-us-central1-ac9805fe-454c-4151-83d0-2daae3aac116 at Monday, 14 November 2022 22:54:33 PDT}
```
Comment (msfschaffner) [inline hw/top_earlgrey/cdc/cdc_waivers.w_fanout.tcl:7]: can you remove the auto-added parts here that contain the smart ID? there are other cases in this PR as well.

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl [hunk 2]
```diff
@@ -4,76 +4,99 @@
...
-set_rule_status -rule {W_DATA} -expression {(ReceivingFlop == "top_earlgrey.u_spi_device.u_reg.u_rxf_ptr_wptr.q[12:0]") && (Signal == "top_earlgrey.u_spi_device.u_fwmode.u_rxf_ctrl.wptr[12:0]") && (Association == "None")} -status {Waived} -lastedit_user {root} -lastedit_time {Monday, 14 November 2022 20:21:17 PDT}
-set_rule_status -rule {W_DATA} -expression {(ReceivingFlop == "top_earlgrey.u_spi_device.u_reg.u_txf_ptr_rptr.q[12:0]") && (Signal == "top_earlgrey.u_spi_device.u_fwmode.u_txf_ctrl.rptr[12:0]") && (Association == "None")} -status {Waived} -lastedit_user {root} -lastedit_time {Monday, 14 November 2022 20:21:17 PDT}
+# W_DATA : included in waived paths
+set_rule_status -rule {W_DATA} -expression {(ReceivingFlop == "top_earlgrey.u_pinmux_aon.mio_oe_retreg_q[46:0]") && (Signal == "top_earlgrey.u_spi_host1.u_cmd_queue.cmd_fifo.gen_normal_fifo.u_fifo_cnt.rptr_o[2:0]") && (Association == "None")} -status {Waived} -comment {included in waived paths}
```
Comment (msfschaffner) [inline hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:8]: for this first batch of waivers regarding the retreg, what has changed? It looks like most of them are identical... is it only the comment that has changed?

if so, it would be good to split these into a separate commit so that it is clear that these are not "new" waivers per se, but just waivers with updated comments (this will make the review quite a bit easier).

**Review by msfschaffner — COMMENTED** [review]
Thanks @jeoongp! Two general comments - I am looking at the new waivers more in detail now.

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 2 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:8]: Yes, only difference is the comment. cdc_waivers.misc.tcl includes additional paths associated with waived paths. I will split this file into a separate commit.

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl [hunk 3]
```diff
@@ -4,76 +4,99 @@
...
+set_rule_status -rule {W_MASYNC} -expression {(ReceivingFlop =~ "top_earlgrey.u_spi_device.u_p2s.out_shift[7:0]") && (MultiClockDomains =~ "IO_DIV2_CLK,IO_DIV4_CLK,SPI_DEV_IN_CLK,SPI_DEV_PASSTHRU_IN_CLK::SPI_DEV_OUT_CLK,SPI_DEV_PASSTHRU_OUT_CLK") && (Driver =~ "top_earlgrey.u_spi_device.u_fwmode.u_tx_fifo.storage[7:0][7:0]")} -status {Waived} -comment {reg data is muxed with spi data in spi clock domain}
+set_rule_status -rule {W_MASYNC} -expression {(ReceivingFlop =~ "top_earlgrey.u_spi_device.u_p2s.out_shift[7:0]") && (MultiClockDomains =~ "IO_DIV4_CLK::SPI_DEV_OUT_CLK,SPI_DEV_PASSTHRU_OUT_CLK") && (Driver =~ "top_earlgrey.u_spi_device.u_reg.u_cfg_tx_order.q[0]")} -status {Waived -comment {reg data is muxed with spi data in spi clock domain}}
+set_rule_status -rule {W_MASYNC} -expression {(ReceivingFlop =~ "top_earlgrey.u_spi_device.u_fwmode.u_rxf_ctrl.sram_wdata[31:0]") && (MultiClockDomains =~ "SPI_DEV_IN_CLK,SPI_DEV_PASSTHRU_IN_CLK::IO_DIV2_CLK,IO_DIV4_CLK,SPI_DEV_IN_CLK,SPI_DEV_PASSTHRU_IN_CLK") && (Driver =~ "top_earlgrey.u_spi_device.u_fwmode.u_rx_fifo.storage[7:0][7:0]")} -status {Waived -comment {reg data is muxed with spi data in spi clock domain}}
```
Comment (msfschaffner) [inline hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:88]: here and in the waiver above: is the syntax correct? it looks like the braces are not closed after {Wavied ...

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl [hunk 4]
```diff
@@ -4,76 +4,99 @@
...
+set_rule_status -rule {W_DATA} -expression {(ReceivingFlop == "top_earlgrey.u_spi_device.u_reg.u_intr_state_generic_rx_watermark.q[0]") && (Signal == "top_earlgrey.u_spi_device.u_fwmode.u_rxf_ctrl.wptr[11:0]") && (Association == "None")} -status {Waived} -comment {included in waived paths}
+set_rule_status -rule {W_DATA} -expression {(ReceivingFlop == "top_earlgrey.u_rv_dm.dap.i_dmi_jtag_tap.tap_state_q[3:0]") && (Signal == "IOR0") && (Association == "None")} -status {Waived}  -comment {included in waived paths}
+set_rule_status -rule {W_DATA} -expression {(ReceivingFlop == "top_earlgrey.u_spi_device.u_reg.u_rxf_ptr_wptr.q[12:0]") && (Signal == "top_earlgrey.u_spi_device.u_fwmode.u_rxf_ctrl.wptr[12:0]") && (Association == "None")} -status {Waived}  -comment {included in waived paths}
+set_rule_status -rule {W_DATA} -expression {(ReceivingFlop == "top_earlgrey.u_spi_device.u_reg.u_txf_ptr_rptr.q[12:0]") && (Signal == "top_earlgrey.u_spi_device.u_fwmode.u_txf_ctrl.rptr[12:0]") && (Association == "None")} -status {Waived}  -comment {included in waived paths}
+
+
+# W_MASYNC : included in waived paths
```
Comment (msfschaffner) [inline hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:83]: @eunchan can you take a look at the SPI ones below in this file?

File: hw/top_earlgrey/cdc/cdc_waivers.w_fanout.tcl [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Verix CDC waiver file
```
Comment (msfschaffner) [inline hw/top_earlgrey/cdc/cdc_waivers.w_fanout.tcl:5]: the `W_FANOUT` waivers in this file LGTM

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 2 above)
Comment (msfschaffner) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:8]: ok - it would also be good if you could split the reformatting of existing waivers and the addition of new waivers into different commits. let me know if you need tips for how to do that with GIT.

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 3 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:88]: Ah. Good catch!

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 2 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl]: I split it and added cdc_waivers.misc.tcl into another commit. Let me know if it looks OK.

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 3 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:95]: Fixed

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl [hunk 6]
```diff
@@ -4,76 +4,106 @@
 #
 # Verix CDC waiver file
```
Comment (msfschaffner) [inline hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:5]: Another question (sorry I don't remember): why were these waivers put into the `*.misc.tcl` file instead of the `*.w_data.tcl` and `*.w_masync.tcl` ones?

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 4 above)
Comment (eunchan) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:90]: yep looks good they are fwmode waivers which the tool confuses the clock information.

Comment (msfschaffner) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:90]: great, thanks for checking!

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 3 above)
Comment (msfschaffner) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:95]: Looks like these are still here (maybe due to the rebase exercise).

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl [hunk 7]
```diff
@@ -4,76 +4,106 @@
...
+set_rule_status -rule {W_DATA} -expression {(ReceivingFlop =~ "top_earlgrey.u_pinmux_aon.mio_oe_retreg_q[46:0]") && (Signal =~ "top_earlgrey.u_spi_host1.u_reg.u_socket.dev_select_outstanding[1:0]") && (Association =~ "None")} -status {Waived}  -comment {included in waived paths}
+set_rule_status -rule {W_DATA} -expression {(ReceivingFlop =~ "top_earlgrey.u_spi_device.u_reg.u_reg_if.rdata_q[3:0]") && (Signal =~ "top_earlgrey.u_spi_device.u_fwmode.u_rx_fifo.sync_wptr.u_sync_2.gen_generic.u_impl_generic.q_o[3:1]") && (Association =~ "None")} -status {Waived}  -comment {included in waived paths}
+set_rule_status -rule {W_DATA} -expression {(ReceivingFlop =~ "top_earlgrey.u_spi_device.u_reg.u_reg_if.rdata_q[2]") && (Signal =~ "top_earlgrey.u_spi_device.u_fwmode.u_rx_fifo.sync_wptr.u_sync_2.gen_generic.u_impl_generic.q_o[0]") && (Association =~ "None")} -status {Waived} -comment {included in waived paths}
```
Comment (msfschaffner) [inline hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:88]: I was curious what the justification `-comment {included in waived paths}` means exactly.
@jeoongp explained to me that these waivers are all paths where the beginning or end points have already been reviewed and excluded as part of a similar related waiver, and they are hence excluded as well.

For example a retention register endpoint in the pinmux where many paths show up as `W_DATA` due to the muxing capability - but not all of them have been excluded yet.

So these LGTM. I asked @jeoongp to expand the comment a bit to make it more clear what `-comment {included in waived paths}` means.

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 6 above)
Comment (msfschaffner) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:5]: Discussed with @jeoongp offline. This is an artifact of the recent restructuring into different files. We decided to leave this for now. @jeoongp is going to clean the structure up a bit more post M2.

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 7 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl]: Updated

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 6 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl:5]: Yes, misc items will be reviewed and added onto w_data and w_masync waivers after M2.

File: hw/top_earlgrey/cdc/cdc_waivers.w_net_no_wave.tcl [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Verix CDC waiver file
+
+set_rule_status -rule {S_NET_NO_WAVE} -expression {(Signal =~ "ast_init_done[3:1]") && (DriverType =~ "Undriven") && (FanoutClockDomain == "IO_DIV4_CLK")} -status {Deferred} -comment {Deferred to Post-M2 while waiting for Nuvoton to clarify}
```
Comment (msfschaffner) [inline hw/top_earlgrey/cdc/cdc_waivers.w_net_no_wave.tcl:7]: Can you file an M3 issue for this and mention the issue number here?

**Review by msfschaffner — APPROVED** [review]

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 7 above)
Comment (msfschaffner) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl]: Thanks for adding that!

File: hw/top_earlgrey/cdc/cdc_waivers.misc.tcl (hunk 3 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.misc.tcl]: Thanks for double check. It's fixed

File: hw/top_earlgrey/cdc/cdc_waivers.w_net_no_wave.tcl (hunk 8 above)
Comment (jeoongp) [reply hw/top_earlgrey/cdc/cdc_waivers.w_net_no_wave.tcl]: Yes, I created #16478 and added it in this comment
